### PR TITLE
prov/efa: Remove some uses of rxr_ep->rdm_ep related to efa_domain

### PR DESCRIPTION
**Pull request by sunkuamzn**
Also add rnr_retry and hmem_p2p_opt to rxr_ep

Signed-off-by: Sai Sunku <sunkusa@amazon.com>

### CODE REVIEW

File: prov/efa/src/efa.h [hunk 1]
```diff
@@ -340,25 +340,10 @@ ssize_t efa_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count, fi_addr_
...
-	if (!rxr_env.use_device_rdma)
-		return 0;
-
-	efa_ep = container_of(ep_fid, struct efa_ep, util_ep.ep_fid);
-	return efa_ep->domain->device->device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ;
-}
-
-static inline
-bool efa_ep_support_rnr_retry_modify(struct fid_ep *ep_fid)
+bool efa_ep_support_rnr_retry_modify(struct efa_domain *domain)
```
Comment (wzamazon) [inline prov/efa/src/efa.h:343]: this function name should be `efa_domain_support_rnr_retry_modify()` and should be moved to `efa_domain.h`

File: prov/efa/src/rxr/rxr.h [hunk 2]
```diff
@@ -728,4 +730,52 @@ static inline void rxr_rm_tx_cq_check(struct rxr_ep *ep, struct util_cq *tx_cq)
...
+/*
+ * @brief: check whether we should use p2p for this transaction
+ *
+ * @param[in]	ep	rxr_ep
+ * @param[in]	efa_mr	memory registration struct
+ *
+ * @return: 0 if p2p should not be used, 1 if it should, and negative FI code
+ * if the transfer should fail.
+ */
+static inline int efa_ep_use_p2p(struct rxr_ep *rxr_ep, struct efa_mr *efa_mr)
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr.h:742]: function name should be `rxr_ep_use_p2p` or `efa_rdm_ep_use_p2p`

File: prov/efa/src/rxr/rxr.h [hunk 3]
```diff
@@ -728,4 +730,52 @@ static inline void rxr_rm_tx_cq_check(struct rxr_ep *ep, struct util_cq *tx_cq)
...
+
+/*
+ * @brief: check whether the endpoint supports rdma read
+ *
+ * @param[in]	domain	struct efa_domain
+ *
+ * @return: true if rdma read is supported. false otherwise.
+ */
+static inline
+bool efa_ep_support_rdma_read(struct efa_domain *domain)
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr.h:773]: function name should be `efa_domain_support_rdma_read` and be moved to `efa_domain.h`

**Review by wzamazon — CHANGES_REQUESTED** [review]

File: prov/efa/src/rxr/rxr.h (hunk 3 above)
Comment (sunkuamzn) [reply prov/efa/src/rxr/rxr.h:773]: needs `rxr_env` which is not available in `efa_domain.h

Comment (wzamazon) [reply prov/efa/src/rxr/rxr.h:773]: add `rxr_env.h`  to `efa_doman.h`?

Comment (sunkuamzn) [reply prov/efa/src/rxr/rxr.h]: done

File: prov/efa/src/rxr/rxr.h (hunk 2 above)
Comment (sunkuamzn) [reply prov/efa/src/rxr/rxr.h]: done

File: prov/efa/src/efa.h (hunk 1 above)
Comment (sunkuamzn) [reply prov/efa/src/efa.h]: done

**Review by wzamazon — APPROVED** [review]

**Pull request merged**